tools/automation.py

```diff
@@ -0,0 +1,316 @@
+AREAS_START_INDEX = 13
+AREAS_END_INDEX = 17
+FOCUS_START_INDEX = 18
+FOCUS_END_INDEX = 22
+PROG_LANG_START_INDEX = 23
+PROG_LANG_END_INDEX = 27
+
+TYPE_AD_HOC = "ad hoc"
+TYPE_LONG_TERM = "long-term"
+TYPE_BOTH = "both"
+IMAGE_FILE_PATH = "assets/images/mentors/"
+IMAGE_SUFFIX = ".jpeg"
+
+class WriteMode(Enum):
+    WRITE = "w"
+    APPEND = "a"
+
+
+def get_social_media_links(*links_args):
+    """
+    Prepare mentor's social media links for yaml network sequence.
+    """
+    network_list = []
+    links_str = ""
+
+    for item in links_args:
+        if not pd.isna(item):
+            links_str += item + " "
+
+    social_media_links_list = links_str.split()
+
+    for link in social_media_links_list:
+        found = 0
+        for name in SOCIAL_MEDIA:
+            if link.find(name) != -1:
+                if name == TELEGRAM_WEB_SITE:
+                    network_list.append({TELEGRAM: link})
+                else:
+                    network_list.append({name: link})
+                found = 1
+                break
+        if found == 0 and len(link):
+            # Use "webpage" as fallback in case of unknown link.
+            network_list.append({WEBSITE: link})
+    return network_list
+
+
+def get_yaml_block_sequence(mentor_data, start_index, end_index):
+    """
+    Yaml block sequence is presented as a list of entries marked with
+    dash and space (“- ”). 
+    """
+    block_sequence_list = []
+    for entry in range(start_index, end_index+1):
+        if not pd.isna(mentor_data.iloc[entry]) and len(mentor_data.iloc[entry].strip()):
+            block_sequence_list.append(str(mentor_data.iloc[entry]).rstrip())
+    return block_sequence_list
+
+
+def extract_numbers_from_string(text_arg, get_max_value=True):
+    """
+    Extract numbers and convert them to integers.
+    """
+    # TODO: If the field hours in xlsx is NAN, what should be the default: 0 or 1
+    if pd.isna(text_arg):
+        text_arg = 0
+
+    if isinstance(text_arg, str):
+        digits = [int(num) for num in re.findall(r"\d+", text_arg)]
+        if digits:
+            if get_max_value:
+                return max(digits)
+            return digits
+
+    return int(text_arg)
+
+
+def get_multiline_string(long_text_arg):
+    """
+    Save strings as yaml multiline strings.
+    Use literal block scalar style to keep newlines (yaml sign: '|-').
+    """
+    multiline_str = ""
+    if not pd.isna(long_text_arg):
+        multiline_str = LiteralScalarString(textwrap.dedent(long_text_arg))
+    return multiline_str
+
+
+def get_mentorship_type(mentorship_type_str):
+    """
+    Returns ad-hoc, long-term, both or empty str
+    """
+    mentorship_type = mentorship_type_str.lower()
+
+    if TYPE_AD_HOC in mentorship_type:
+        return TYPE_AD_HOC.replace(' ', '-')
+    elif TYPE_LONG_TERM in mentorship_type:
+        return TYPE_LONG_TERM
+    elif TYPE_BOTH in mentorship_type:
+        return TYPE_BOTH
+
+    return ""
```

Comment:
```suggestion
    return "NOT_FOUND"
```

or undefined, just to be easier to find in case that happen.

Reply:
Done.